python/lsst/daf/butler/remote_butler/_remote_butler.py

```diff
@@ -545,9 +560,53 @@ def _get_url(self, path: str, version: str = "v1") -> str:
         return f"{version}/{path}"
 
     def _post(self, path: str, model: _BaseModelCompat) -> httpx.Response:
+        """Send a POST request to the Butler server."""
         json = model.model_dump_json(exclude_unset=True).encode("utf-8")
         url = self._get_url(path)
         return self._client.post(url, content=json, headers={"content-type": "application/json"})
 
+    def _get(self, path: str) -> httpx.Response:
+        """Send a GET request to the Butler server."""
+        url = self._get_url(path)
+        return self._client.get(url)
+
     def _parse_model(self, response: httpx.Response, model: Type[_AnyPydanticModel]) -> _AnyPydanticModel:
+        """Deserialize a Pydantic model from the body of an HTTP response."""
         return model.model_validate_json(response.content)
+
+    def _normalize_collections(self, collections: str | Sequence[str] | None) -> CollectionList:
+        """Convert the ``collections`` parameter in the format used by Butler
+        methods to a standardized format for the REST API.
+        """
+        if collections is None:
+            if not self.collections:
+                raise NoDefaultCollectionError(
+                    "No collections provided, and no defaults from butler construction."
+                )
+            collections = self.collections
+        # Temporary hack. Assume strings for collections. In future
+        # want to construct CollectionWildcard and filter it through collection
+        # cache to generate list of collection names.
+        wildcards = CollectionWildcard.from_expression(collections)
+        return CollectionList(list(wildcards.strings))
+
+    def _normalize_dataset_type_name(self, datasetTypeOrName: DatasetType | str) -> DatasetTypeName:
+        """Convert DatasetType parameters in the format used by Butler methods
+        to a standardized string name for the REST API.
+        """
+        if isinstance(datasetTypeOrName, DatasetType):
```

Comment:
Would 

```python
return getattr(datasetTypeOrName, "name", datasetTypeOrName)
```
be more efficient?

Reply:
Slightly, yeah, but less type safe and more ambiguous.  I personally have no idea off the top of my head whether `str` or any str-like objects that may be passed in by a user have a `name` attribute or not.